cw-relayer/config/config.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/spf13/viper"
+)
+
+const (
+	defaultListenAddr      = "0.0.0.0:7171"
+	defaultProviderTimeout = 100 * time.Millisecond
+	//TODO: add default ojo rpc
+	defaultQueryRPC        = ""
+	defaultMissedThreshold = 5
+)
+
+var (
+	validate = validator.New()
+
+	// ErrEmptyConfigPath defines a sentinel error for an empty config path.
+	ErrEmptyConfigPath = errors.New("empty configuration file path")
+)
+
+type (
+	// Config defines all necessary cw-relayer configuration parameters.
+	Config struct {
+		Account         Account `mapstructure:"account" validate:"required,gt=0,dive,required"`
+		Keyring         Keyring `mapstructure:"keyring" validate:"required,gt=0,dive,required"`
+		RPC             RPC     `mapstructure:"rpc" validate:"required,gt=0,dive,required"`
+		GasAdjustment   float64 `mapstructure:"gas_adjustment" validate:"required"`
+		ContractAddress string  `mapstructure:"contract_address"`
+		ProviderTimeout string  `mapstructure:"provider_timeout"`
+		MissedThreshold int64   `mapstructure:"missed_threshold"`
+		// query rpc for ojo node
+		QueryRPC string `mapstructure:"query_rpc"`
+	}
+
+	// Account defines account related configuration that is related to the Client
+	// Network and Receives Pricing information.
+	Account struct {
+		ChainID string `mapstructure:"chain_id" validate:"required"`
+		Address string `mapstructure:"address" validate:"required"`
+	}
+
+	// Keyring defines the required Client-chain keyring configuration.
+	Keyring struct {
+		Backend string `mapstructure:"backend" validate:"required"`
+		Dir     string `mapstructure:"dir" validate:"required"`
+	}
+
+	// RPC defines RPC configuration of both the wasmd chain gRPC and Tendermint nodes.
+	RPC struct {
+		TMRPCEndpoint string `mapstructure:"tmrpc_endpoint" validate:"required"`
+		GRPCEndpoint  string `mapstructure:"grpc_endpoint" validate:"required"`
+		RPCTimeout    string `mapstructure:"rpc_timeout" validate:"required"`
+	}
+
+	MsgRelay struct {
+		Relay Msg `json:"relay"`
+	}
+
+	MsgForceRelay struct {
+		Relay Msg `json:"force_relay"`
+	}
+
+	Msg struct {
+		SymbolRates [][2]string `json:"symbol_rates,omitempty"`
+		ResolveTime int64       `json:"resolve_time,omitempty"`
+		RequestID   uint64      `json:"request_id,omitempty"`
```

Comment:
These types shouldn't be in the `config` package. Let's put them in a file called `messages.go` in cw-relayer/relayer